lightning/src/ln/channelmanager.rs

```diff
@@ -3405,6 +3398,61 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		for (source, payment_hash, reason) in timed_out_htlcs.drain(..) {
 			self.fail_htlc_backwards_internal(self.channel_state.lock().unwrap(), source, &payment_hash, reason);
 		}
+	}
+
+	/// Updates channel state to take note of transactions which were confirmed in the given block
+	/// at the given height.
+	///
+	/// Note that you must still call update_best_block with the block information which is
+	/// included here.
+	///
+	/// This method may be called before or after update_best_block for a given block's transaction
```

Comment:
I think it just depends on the exact API calls to the block source you have/use. It may notify you of the block changing before or after the transactions, but the point is we don't care either way.